doc/howtos/pull-request-guidelines.rst

```diff
@@ -0,0 +1,161 @@
+might get overlooked. A submitter can use this to avoid these mistakes
+before even submitting the pull request.
+
+Workflow
+========
+
+PRs are typically checked in this order:
+
+1. pull request message
+2. commit messages
+3. actual diff
+
+Reviewers, beware that jumping straight to a diff may miss relevant
+explanations.
+
+Submitters, reviewers are typically busy. If there are issues with
+the PR, a review might get aborted already at an early stage with
+a request to fix those things first before the review continues.
+
+Pull Request Message
+====================
+
+The main pull request message summarizes the change. Relevant
+information is how important the PR is, whether there are dependencies
+on other PRs, and who should, might or needs to review it.
+
+Submitters, ideally mention reviewers using their Github handle with
+the at symbol to alert them of the PR, as in ``@johndoe please
+review`` or ``@joandoe needs to review this``.
+
+Commit Messages
+===============
+
+Should follow the format explained in :ref:`contributor-guide`, for
+the sake of consistency and completeness.
+
+The commit message will be used in different ways and with different
+goals:
+
+* Listing commits (``git log --oneline``, Jenkins PR overview pages, ``gitk``):
+  the first line should be informative and ideally unique.
+* Reviewing the change: what is the motivation for the change, why
+  was it solved this way?
+* Distro maintenance at a later point in time: is the change still relevant?
+  What breaks if the change has to removed or modified? This might also
+  get answered in the actual code, but sometimes there is additional information
+  like verbatim copies of error messages that are too large for the code
+  itself. Storing them in the commit message preserves that additional
+  information (see next point).
+* Searching for problems and how they were solved: ``git log --grep`` can be
+  useful to find commits again.
+* Commit author: when including something written by others (like patches
+  or complete files), try to submit that in a separate commit with the
+  original author set as author of the commit. This is important for tracking
+  the origin of code in Ostro OS, which partly is done by automated tooling.
+  When multiple people authored the change, use the upstream project's
+  mailing list as author. Alternatively, using ``git am`` format for patches
+  can also be used to document the author of a code patch (see next section).
+
+
+Content of a Pull Request
+=========================
+
+When a change affects user-visible behavior, make sure that relevant
+user documentation gets updated, ideally in the same commit (only
+works when submitting to ``meta-ostro``, which is where the ``doc``
+directory is maintained). If that is not possible or not desirable at
+the time, include enough information in the change or commit message
+for a technical writer to update the Ostro OS user documentation
```

Comment:
It would be nice if we had a way to mark such changes, but currently we don't have anything defined.

Perhaps we should have a special "TODO DOC: do something" convention for comments in the source code which then can be searched for with "git grep 'TODO DOC'" in the ostro-os repository?